### ISSUE TITLE
Allow user to provide a GH token in install scripts 

### ISSUE THREAD

**Issue by jribbink** [onflow/flow-cli#1031]
## Instructions

Please fill out the template below to the best of your ability.

### Issue To Be Solved

We should allow the user to provide a GH token in the install scripts.  In some cases, the user may be rate limited if they are installing the CLI frequently (i.e. when being used in CI/CD pipeline)

### (Optional): Suggest A Solution

Add parameters to scripts which add this to the cURL call.

### (Optional): Context

Blocking me in Cadence VSCode extension.  Windows tests are failing CI because I am getting rate limited by GitHub.

**Comment by sideninja** [comment]
Do tests download the binary only once?

**Comment by jribbink** [comment]
> Is the CLI binary downloaded only once when tests are run? if so, it seems a bit weird you would get rate limited.

Yeah.  I'm not sure how the whole server/IP sharing situation is with GH actions (is it that a lot of people's CI tests from the same IP access github.com in a given hour => rate limited?).

The CI run can be found here: https://github.com/onflow/vscode-cadence/actions/runs/5018043534/jobs/8996942771#step:5:41

**Comment by jribbink** [comment]
Just checked and the rate limit is 60 req/hour for unauthenticated users
